### ISSUE TITLE
map, vmap problems

### ISSUE THREAD

**Issue by firelook** [TrinityCore/TrinityCore#25501]
Windows 10
Wow French Version
version 35662
when I extract data files using your tools, and put it in the worldserver to be used I have : 

`Could not load VMAP name:Royaumes de l'Est, id:0, x:38, y:33 (vmap rep.: x:38, y:33)
MMAP:loadMap: Could not load 22923833.mmtile into navmesh
Could not load VMAP name:Royaumes de l'Est, id:0, x:48, y:29 (vmap rep.: x:48, y:29)
MMAP:loadMap: Could not load 22924829.mmtile into navmesh
Could not load VMAP name:Royaumes de l'Est, id:0, x:27, y:30 (vmap rep.: x:27, y:30)
MMAP:loadMap: Could not load 22922730.mmtile into navmesh
Could not load VMAP name:Royaumes de l'Est, id:0, x:39, y:32 (vmap rep.: x:39, y:32)
MMAP:loadMap: Could not load 17233932.mmtile into navmesh
MMAP:loadMap: Could not load 22923932.mmtile into navmesh
Could not load VMAP name:Royaumes de l'Est, id:0, x:55, y:30 (vmap rep.: x:55, y:30)
Could not load VMAP name:Norfendre, id:571, x:28, y:43 (vmap rep.: x:28, y:43)
Could not load VMAP name:Kalimdor, id:1, x:17, y:30 (vmap rep.: x:17, y:30)
Could not load VMAP name:Norfendre, id:571, x:32, y:42 (vmap rep.: x:32, y:42)
Could not load VMAP name:Royaumes de l'Est, id:0, x:45, y:27 (vmap rep.: x:45, y:27)
MMAP:loadMap: Could not load 22924527.mmtile into navmesh
Could not load VMAP name:Kalimdor, id:1, x:39, y:41 (vmap rep.: x:39, y:41)
MMAP:loadMap: Could not load 17493941.mmtile into navmesh
Could not load VMAP name:Kalimdor, id:1, x:25, y:43 (vmap rep.: x:25, y:43)
MMAP:loadMap: Could not load 17492543.mmtile into navmesh
Could not load VMAP name:Kalimdor, id:1, x:16, y:29 (vmap rep.: x:16, y:29)
MMAP:loadMap: Could not load 17491629.mmtile into navmesh
Could not load VMAP name:Kalimdor, id:1, x:28, y:40 (vmap rep.: x:28, y:40)
MMAP:loadMap: Could not load 17492840.mmtile into navmesh
Could not load VMAP name:Kalimdor, id:1, x:28, y:43 (vmap rep.: x:28, y:43)
Could not load VMAP name:Kalimdor, id:1, x:34, y:39 (vmap rep.: x:34, y:39)
MMAP:loadMap: Could not load 17493439.mmtile into navmesh
Could not load VMAP name:Royaumes de l'Est, id:0, x:28, y:30 (vmap rep.: x:28, y:30)
MMAP:loadMap: Could not load 22922830.mmtile into navmesh
Could not load VMAP name:Norfendre, id:571, x:27, y:19 (vmap rep.: x:27, y:19)
`

please what is my mistake ?

**Comment by jackpoz** [comment]
We set an issue template when creating issues, please don't ignore the template

**Comment by firelook** [comment]
I try o put the most relevant
branch : Master
version : latest
no mods
wow version 8

**Comment by Faq** [comment]
There is no thing like "latest". Please put exact hash string

**Comment by firelook** [comment]
I do it daily when you an update on your side
so my version today is as of : 
`Author: Shauren <shauren.trinity@gmail.com>  2020-09-25 19:54:37
Committer: Shauren <shauren.trinity@gmail.com>  2020-09-25 19:54:37
Parent: f922c6e7a46a7c712daaaef9b1a72ca865fbe94b (Core/PacketIO: Renamed a bunch of opcodes based on more research (only those added after 6.0))
Branches: master, remotes/origin/master
Follows: TDB837.20081
Precedes: 

    Core: Add mac client support for build 35662`

**Comment by Destructor** [comment]
Met ton jeu en anglais et fait ton extraction, tu auras plus ce problème, je pense qu'il y a un bug au niveau des noms français que l'extracteur n'arrive pas a comprendre (genre les accents et autres ponctuations).
La de mon coté je suis entrain de tout extraire et j'ai pas eu une erreur alors qu'avec la version française certaines maps etaient en erreur.

**Comment by Destructor** [comment]
@firelook

**Comment by illfated** [comment]
Could this be caused by incomplete game client, i.e. only partially downloaded map data in the game client?

**Comment by firelook** [comment]
I am using the "official " battlenet launcher to maintain this application

**Comment by illfated** [comment]
Fair enough, thank you for the reply. Maybe the suggestion from @Destructor is a usable workaround until a real solution can be found:
> Put your game in English and do your extraction, you will no longer have this problem, I think there is a bug with the French names that the extractor cannot understand (like accents and other punctuation). On my side I am extracting everything and I did not have an error while with the French version some maps were in error.

**Comment by firelook** [comment]
I have erased the directory, switch the language to english, download the client et re extract everything
I have still a huge list.
https://pastebin.com/svi5UY57

**Comment by illfated** [comment]
I hope someone else may be able to help you with further troubleshooting of this issue.
I don't have a master branch setup because I am focused on the 3.3.5 branch.
I was only hoping that it would be a simpler solution than this.
I wish you the best of luck going forward.

**Comment by firelook** [comment]
thanks, I do not know honestly what to do

**Comment by illfated** [comment]
The only suggestions I have, is to troubleshoot for typical user configuration issues, like incorrectly placed map folders (compared to worldserver.conf setup) and, if possible, talk with someone you know who uses the same server setup and game client as you, to compare results.

If your contact has fewer errors than you (and possibly not the same map issues), that would suggest a user configuration problem.
If enough users can confirm the same issue as you, using unmodified source & game client, it is more likely to be a source code issue.

**Comment by firelook** [comment]
in the dbc extract, I have this error : 
`______                       __
/\__  _\       __          __/\ \__
\/_/\ \/ _ __ /\_\    ___ /\_\ \, _\  __  __
   \ \ \/\`'__\/\ \ /' _ `\/\ \ \ \/ /\ \/\ \
    \ \ \ \ \/ \ \ \/\ \/\ \ \ \ \ \_\ \ \_\ \
     \ \_\ \_\  \ \_\ \_\ \_\ \_\ \__\\/`____ \
      \/_/\/_/   \/_/\/_/\/_/\/_/\/__/ `/___/> \
                                 C O R E  /\___/
http://TrinityCore.org                    \/__/

Opened casc storage 'E:/games\World of Warcraft\Data'
Opened casc storage 'E:/games\World of Warcraft\Data'
Detected client build 35662 for locale enUS

Extracting dbc/db2 files...
locale enUS output path E:\games\World of Warcraft\dbc\enUS
Unable to open file SSAOSettings.db2 in the archive for locale enUS: FILE_NOT_FOUND
Extracted 715 files

Opened casc storage 'E:/games\World of Warcraft\Data'
Detected client build 35662 for locale frFR

Extracting dbc/db2 files...
locale frFR output path E:\games\World of Warcraft\dbc\frFR
Unable to open file SSAOSettings.db2 in the archive for locale frFR: FILE_NOT_FOUND
Extracted 715 files`

is it a normal error please ?

**Comment by illfated** [comment]
Sorry, I don't know. We must wait for others who are using master branch and a French (frFR) version of the game to see if they get this type of issue, or if any of the real coders have any idea what happens with your extraction process.

**Comment by shdr212** [comment]
I'm having the same issue. Not sure if it actually causes anything yet, but the warnings are also showing up.

So far the only warnings that shows up are the WDT ones, such as:
```
Can't open WDT for map 451 - Development Land (FileDataID 857684): FILE_NOT_FOUND
```

**Comment by illfated** [comment]
I am fairly certain "Development Land" is one of the less important issues, since it is not really part of the standard game content.

**Comment by shdr212** [comment]
@illfated It was just an example. There were many other without that extra "suffix".

**Comment by illfated** [comment]
> @illfated It was just an example. There were many other without that extra "suffix".

Fair enough. I don't know the solution to this, so can only I hope it will be resolved soon™.

**Comment by firelook** [comment]
I try to re download different version, re extract  
I can not enter any world because I have this error now : 
`TrinityCore rev. 2e4609f6e18c+ 2020-12-27 22:43:36 +0100 (master branch) (Win64, Release, Static) (worldserver-daemon) ready...
Received not handled opcode [CMSG_BATTLE_PAY_GET_PURCHASE_LIST 0x36BC (14012)] from [Player: Account: 23]
Received not handled opcode [CMSG_BATTLE_PAY_GET_PRODUCT_LIST 0x36BB (14011)] from [Player: Account: 23]
Received not handled opcode [CMSG_UPDATE_VAS_PURCHASE_STATES 0x36F5 (14069)] from [Player: Account: 23]
Received not handled opcode [CMSG_GET_ACCOUNT_CHARACTER_LIST 0x36B6 (14006)] from [Player: Account: 23]
Received not handled opcode [CMSG_GET_RAF_ACCOUNT_INFO 0x3723 (14115)] from [Player: Account: 23]
Received not handled opcode [CMSG_GET_REMAINING_GAME_TIME 0x36ED (14061)] from [Player: Account: 23]
Received not handled opcode [CMSG_QUICK_JOIN_AUTO_ACCEPT_REQUESTS 0x3708 (14088)] from [Player: Account: 23]
Received not handled opcode [CMSG_BATTLE_PAY_GET_PURCHASE_LIST 0x36BC (14012)] from [Player: Account: 23]
Could not load MMAP name:Royaumes de l’Est, id:0, x:58, y:30 (mmap rep.: x:58, y:30)
Could not load MMAP name:Kalimdor, id:1, x:17, y:30 (mmap rep.: x:17, y:30)
Could not load MMAP name:Royaumes de l’Est, id:0, x:36, y:32 (mmap rep.: x:36, y:32)`

I suspect I need now a good mmap directory but I have no clues where to look, any suggestions/ideas  please ?

**Comment by illfated** [comment]
@firelook : Hmm. I remember seeing those opcodes long ago, when I was still using and testing the master branch on 6.x and 7.x, but those opcode errors did not hinder me from logging in. Have you tried comparing with the enUS game client, to see if the opcodes are resolved there or are still causing the same errors? Also, just to make sure, are you using the Arctium game client launcher?

**Comment by firelook** [comment]
I do not think op code is the main issue
I suspect the main issues is I do not have some mmaps

**Comment by illfated** [comment]
From the main ticket comment, on Sep 26 (https://github.com/TrinityCore/TrinityCore/issues/25501#issue-709561753):

> Wow French Version
> version 35662
> when I extract data files using your tools, and put it in the worldserver to be used I have :
>
> ```
> Could not load VMAP name:Royaumes de l'Est, id:0, x:38, y:33 (vmap rep.: x:38, y:33) 
> MMAP:loadMap: Could not load 22923833.mmtile into navmesh Could not load VMAP name:Royaumes de l'Est, id:0, x:48, y:29 (vmap rep.: x:48, y:29)
> MMAP:loadMap: Could not load 22924829.mmtile into navmesh Could not load VMAP name:Royaumes de l'Est, id:0, x:27, y:30 (vmap rep.: x:27, y:30)
> MMAP:loadMap: Could not load 22922730.mmtile into navmesh Could not load VMAP name:Royaumes de l'Est, id:0, x:39, y:32 (vmap rep.: x:39, y:32)
> MMAP:loadMap: Could not load 17233932.mmtile into navmesh
> MMAP:loadMap: Could not load 22923932.mmtile into navmesh Could not load VMAP name:Royaumes de l'Est, id:0, x:55, y:30 (vmap rep.: x:55, y:30)
> Could not load VMAP name:Norfendre, id:571, x:28, y:43 (vmap rep.: x:28, y:43)
> Could not load VMAP name:Kalimdor, id:1, x:17, y:30 (vmap rep.: x:17, y:30)
> Could not load VMAP name:Norfendre, id:571, x:32, y:42 (vmap rep.: x:32, y:42)
> Could not load VMAP name:Royaumes de l'Est, id:0, x:45, y:27 (vmap rep.: x:45, y:27)
> MMAP:loadMap: Could not load 22924527.mmtile into navmesh Could not load VMAP name:Kalimdor, id:1, x:39, y:41 (vmap rep.: x:39, y:41)
> MMAP:loadMap: Could not load 17493941.mmtile into navmesh Could not load VMAP name:Kalimdor, id:1, x:25, y:43 (vmap rep.: x:25, y:43)
> MMAP:loadMap: Could not load 17492543.mmtile into navmesh Could not load VMAP name:Kalimdor, id:1, x:16, y:29 (vmap rep.: x:16, y:29)
> MMAP:loadMap: Could not load 17491629.mmtile into navmesh Could not load VMAP name:Kalimdor, id:1, x:28, y:40 (vmap rep.: x:28, y:40)
> MMAP:loadMap: Could not load 17492840.mmtile into navmesh Could not load VMAP name:Kalimdor, id:1, x:28, y:43 (vmap rep.: x:28, y:43)
> Could not load VMAP name:Kalimdor, id:1, x:34, y:39 (vmap rep.: x:34, y:39)
> MMAP:loadMap: Could not load 17493439.mmtile into navmesh Could not load VMAP name:Royaumes de l'Est, id:0, x:28, y:30 (vmap rep.: x:28, y:30)
> MMAP:loadMap: Could not load 22922830.mmtile into navmesh Could not load VMAP name:Norfendre, id:571, x:27, y:19 (vmap rep.: x:27, y:19)
> ```
>
> please what is my mistake ?
>

I don't know what the exact issue is, but it looks like either the game client could be incomplete (not having all the required data or files), or the TC extractors are not able to read and extract data from the frFR client.
I can only suggest to troubleshoot the game client, checking the content by using a test account on retail to see if it is possible to visit all locations available to new characters.

***

From the comment on Oct 17 (https://github.com/TrinityCore/TrinityCore/issues/25501#issuecomment-710744698):

> in the dbc extract, I have this error :
>
> ```
>  ______                       __
> /\__  _\       __          __/\ \__
> \/_/\ \/ _ __ /\_\    ___ /\_\ \, _\  __  __
>    \ \ \/\`'__\/\ \ /' _ `\/\ \ \ \/ /\ \/\ \
>     \ \ \ \ \/ \ \ \/\ \/\ \ \ \ \ \_\ \ \_\ \
>      \ \_\ \_\  \ \_\ \_\ \_\ \_\ \__\\/`____ \
>       \/_/\/_/   \/_/\/_/\/_/\/_/\/__/ `/___/> \
>                                  C O R E  /\___/
> http://TrinityCore.org                    \/__/
> ```
> 
> Opened casc storage 'E:/games\World of Warcraft\Data'
> Opened casc storage 'E:/games\World of Warcraft\Data'
> Detected client build 35662 for locale enUS
> 
> Extracting dbc/db2 files...
> locale enUS output path E:\games\World of Warcraft\dbc\enUS
> Unable to open file SSAOSettings.db2 in the archive for locale enUS: FILE_NOT_FOUND
> Extracted 715 files
> 
> Opened casc storage 'E:/games\World of Warcraft\Data'
> Detected client build 35662 for locale frFR
> 
> Extracting dbc/db2 files...
> locale frFR output path E:\games\World of Warcraft\dbc\frFR
> Unable to open file SSAOSettings.db2 in the archive for locale frFR: FILE_NOT_FOUND
> Extracted 715 files`

"Unable to open file SSAOSettings.db2" is either a sign of an incomplete game client or, if confirmed, a file that is not used on the server side for extraction.
Unfortunately, I don't know the 9.x client well enough to tell which one.

> and some samples errors in the map extract
>
> ```
> Can't open WDT for map 1964 - PhaseableMO: War Campaign - Horde - Chapter 05 - Kul Tiras Flagship (FileDataID 2449171): FILE_NOT_FOUND
> Can't open WDT for map 1965 - PhaseableMO: War Campaign - Horde - Chapter 05 - Kul Tiras Ship (FileDataID 2449176): FILE_NOT_FOUND
> Processing Map 2066 [################################################################]
> Can't open WDT for map 2067 - Zandalari Battleship - Gral's Call (FileDataID 2449194): FILE_NOT_FOUND
> Can't open WDT for map 2068 - [Not Used] (FileDataID 2449193): FILE_NOT_FOUND
> Can't open WDT for map 2069 - Zuldazar Alliance Bombing Run (FileDataID 2449189): FILE_NOT_FOUND
> Processing Map 2070 [################################################################]
> Can't open WDT for map 2075 - 8.0 Islands - Event (FileDataID 2449190): FILE_NOT_FOUND
> Processing Map 2076 [################################################################]
> Can't open WDT for map 2077 - 8.0 Islands - Event (FileDataID 2449191): FILE_NOT_FOUND
> Can't open WDT for map 2078 - 8.0 Islands - Event (FileDataID 2449192): FILE_NOT_FOUND
> Processing Map 2081 [################################################################]
> Can't open WDT for map 2082 - 8.0 Islands - Micro - Mogu Crypt (FileDataID 2449195): FILE_NOT_FOUND
> Can't open WDT for map 2083 - 8.0 Islands - Micro - Mogu Crypt (FileDataID 2449196): FILE_NOT_FOUND
> Can't open WDT for map 2084 - 8.0 Islands - Micro - Mogu Crypt (FileDataID 2449197): FILE_NOT_FOUND
> Can't open WDT for map 2085 - 8.0 Islands - Micro - Mogu Crypt (FileDataID 2449198): FILE_NOT_FOUND
> Can't open WDT for map 2086 - 8.0 Islands - Micro - Mogu Crypt (FileDataID 2449199): FILE_NOT_FOUND
> Can't open WDT for map 2087 - 8.0 Islands - Micro - Mogu Crypt (FileDataID 2449200): FILE_NOT_FOUND
> Can't open WDT for map 2088 - 8.0 Islands - Micro - Mogu Crypt (FileDataID 2449201): FILE_NOT_FOUND
> ```
>
> is it a normal error please ?

Those errors are much more likely to be a sign of an incomplete game client.
But, in theory, it could also be some kind of issue with the map extractors not being able to read the frFR content as well as the enUS content.

I wish I could have known more, but becauseI am not using the master branch and the 9.x client, it is difficult for me to find out.
I can only hope someone using non-English game clients could post comments and tell us what their experience is.

**Comment by illfated** [comment]
> by looking the logs, I have found I hope some help
> `Loading Transports...
> 
> Could not load MMAP name:Royaumes de l’Est, id:0, x:28, y:31 (mmap rep.: x:28, y:31)
> Could not load MMAP name:Royaumes de l’Est, id:0, x:31, y:37 (mmap rep.: x:31, y:37)

Messages like this can usually be ignored when they are found in Server.log (and on the console). But I am not 100% certain if this should be a problem when seen in the extraction process.

**Comment by firelook** [comment]
Due to the fact I can not connect, it hangs at the login at 80% and I see this I can suspect it is a problem

P.S.
obvisouly I downloaded from scratch, different version, .... to try to re extract all data (vmaps, mmaps,...)

**Comment by illfated** [comment]
> Due to the fact I can not connect, it hangs at the login at 80% and I see this I can suspect it is a problem
> 
> P.S.
> obvisouly I downloaded from scratch, different version, .... to try to re extract all data (vmaps, mmaps,...)

Fair enough. If you don't get any useful feedback in this issue ticket, try asking in Discord / IRC.

**Comment by Aokromes** [comment]
this bug is still valid?

**Comment by Symbolixdevic** [comment]
Yes 
![Uploading изображение_2024-06-07_230056999.png…]()

**Comment by shamage** [comment]
I performed a test in the form of switching the game to the local French language + tested whether the game really works for me in this language.

Subsequently, I performed data extraction with the help of tools.


```sql
TrinityCore rev. 53cfa0dbb1c7 2024-06-05 19:27:10 +0200 (main branch) (Win64, RelWithDebInfo, Static) (Map & DBC Extractor)
<Ctrl-C> to stop.

 ______ __
/\__ _\ __ __/\ \__
\/_/\ \/ _ __ /\_\ ___ /\_\ \, _\ __ __
 \ \ \/\`'__\/\ \ /' _ `\/\ \ \ \/ /\ \/\ \
 \ \ \ \ \/ \ \ \/\ \/\ \ \ \ \ \_\ \ \_\ \
 \ \_\ \_\ \ \_\ \_\ \_\ \_\ \__\\/`____ \
 \/_/\/_/ \/_/\/_/\/_/\/_/\/__/ `/___/> \
 C O R E /\___/
http://TrinityCore.org \/__/

Opened casc storage 'I:\World of Warcraft\Data'
Opened casc storage 'I:\World of Warcraft\Data'
Detected client build 54988 for locale enUS

Extracting dbc/db2 files...
locale enUS output path I:\World of Warcraft\dbc\enUS
Extracted 987 files

Opened casc storage 'I:\World of Warcraft\Data'
Detected client build 54988 for locale frFR

Extracting dbc/db2 files...
locale frFR output path I:\World of Warcraft\dbc\frFR
Extracted 987 files

Opened casc storage 'I:\World of Warcraft\Data'
Detected client build 54988 for local deDE

Extracting dbc/db2 files...
locale deDE output path I:\World of Warcraft\dbc\deDE
Extracted 987 files

Opened casc storage 'I:\World of Warcraft\Data'
Extracting camera files...
Read CinematicCamera.db2 file...
Done! (27 CinematicCameras loaded)
output path I:\World of Warcraft\cameras
Extracted 27 camera files
Opened casc storage 'I:\World of Warcraft\Data'
Extracting game tables...
output path I:\World of Warcraft\gt
Extracted 20 files

Opened casc storage 'I:\World of Warcraft\Data'
Extracting maps...
Read Map.db2 file...
Done! (938 maps loaded)
Read LiquidMaterial.db2 file...
Done! (11 LiquidMaterials loaded)
Read LiquidObject.db2 file...
Done! (4941 LiquidObjects loaded)
Read LiquidType.db2 file...
Done! (173 LiquidTypes loaded)
Convert map files
Extract Eastern Kingdoms (1/938)
Extract Kalimdor (2/938)
Extract Alterac Valley (3/938)
Extract Shadowfang Keep (4/938)
Extract Stormwind Stockade (5/938)
Extract Deadmines (6/938)
Extract Wailing Caverns (7/938)
Extract Razorfen Kraul (8/938)
Extract Blackfathom Deeps (9/938)
Extract Uldaman (10/938)

////////////////////////////////////////////////////////////////////////////

[Map 2695] [39,32]:  Adding tile to navmesh...
[Map 2695] [39,32]:  Writing to file...
[Map 2695] [40,38]:  Building navmesh tile...
[Map 2695] [40,38]:  Adding tile to navmesh...
[Map 2695] [40,38]:  Writing to file...
[Map 2695] [40,37]:  Building navmesh tile...
[Map 2695] [40,37]:  Adding tile to navmesh...
[Map 2695] [40,37]:  Writing to file...
[Map 2695] [39,35]:  Building navmesh tile...
[Map 2695] [39,35]:  Adding tile to navmesh...
[Map 2695] [39,35]:  Writing to file...
[Map 2695] [41,37]:  Building navmesh tile...
[Map 2695] [41,37]:  Adding tile to navmesh...
[Map 2695] [41,37]:  Writing to file...
Finished. MMAPS were built in 50 Minutes 39 Seconds.

...............................................
Trinitycore dbc/db2, maps, vmaps, mmaps extractor
...............................................
PRESS 1, 2, 3 OR 4 to select your task, or 5 to EXIT.
...............................................

WARNING! when extracting the vmaps extractor will
output the textS below, it's intended and not an error:
..........................................
Extracting World\Wmo\Band\Final_Stage.wmo
No such file.
Couldn't open RootWmo!!!
Done!
..........................................
The system cannot find the file specified.
..........................................

Press 1, 2, 3 or 4 to start extracting or 5 to exit.
1 - Extract base files (NEEDED) and cameras.
2 - Extract vmaps (needs maps to be extracted before you run this) (OPTIONAL, highly recommended)
3 - Extract mmaps (needs vmaps to be extracted before you run this, may take hours) (OPTIONAL, highly recommended)
4 - Extract all (may take hours)
5 - EXIT

Type 1, 2, 3, 4 or 5 then press ENTER:
```

The entire extraction process went without errors.

I am not sure if my procedure was correct for the local language (other than English) or if the extraction procedure needs to be adjusted differently...

If it has to be done differently - I will be happy if someone provides us with the data / procedures and I can do another test accordingly. If my procedure was correct, the error will not be in Core but probably in the user.

**Comment by Aokromes** [comment]
this bug is still valid?
